### PR TITLE
#104 - Add XmlPackagesFile

### PR DESCRIPTION
**Pull request by HDouss**
Resolving issue #104 
Added a puzzle for adding additional namespaces in xml.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#142](https://github.com/artipie/rpm-adapter/issues/142) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/879b1ed7-9f46-4c05-ac5b-b25a0a971a27); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by HDouss** [comment]
@Vatavuk ping

File: src/test/java/com/artipie/rpm/meta/XmlPackagesFileTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Test
+    public void writesCorrectTag(@TempDir final Path temp) throws Exception {
+        final Path file = temp.resolve("tag.xml");
+        final String expected = "thetag";
+        try (XmlPackagesFile packs = new XmlPackagesFile(new XmlFile(file), expected, "")) {
+            packs.startPackages();
+        }
+        MatcherAssert.assertThat(
+            new String(Files.readAllBytes(file), StandardCharsets.UTF_8),
```
Comment (Vatavuk) [inline src/test/java/com/artipie/rpm/meta/XmlPackagesFileTest.java:50]: @HDouss this can be extracted to a private method to avoid code repetition

**Comment by HDouss** [comment]
@Vatavuk Thanks. Fixed!

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@dc9a75c2-99a9-40a1-bc27-ddaf8f08c10c, thread: pool-910-thread-1
java.lang.Thread[-2] java.lang.OutOfMemoryError: unable to create new native thread
```

[0.53.12](https://github.com/zerocracy/farm): Issue: artipie/rpm-adapter#142, Comment: 619908907

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@dc9a75c2-99a9-40a1-bc27-ddaf8f08c10c, thread: pool-910-thread-1
com.zerocracy.radars.github.ReSafe[81] java.io.IOException: java.lang.OutOfMemoryError: unable to create new native thread
java.lang.Thread[-2] java.lang.OutOfMemoryError: unable to create new native thread
```

[0.53.12](https://github.com/zerocracy/farm): Issue: artipie/rpm-adapter#142, Action: `created`

**Comment by Vatavuk** [comment]
@sereshqua please try again

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**